### DISCUSSION TITLE
Proxy is not supported with headless template

### DISCUSSION THREAD

**Discussion by tomdottom** [projectdiscovery/nuclei#1113] [Q&A] [answered]
Has anyone successfully used the proxy configuration in combination with a headless test?

I've found that normal templates can use the proxy config just fine, but headless templates using Chromium do not.

From what I can tell the proxy setting is passed to the browser in code:

https://github.com/projectdiscovery/nuclei/blob/f5fb8aa305121ceb28981436008acb180b99f0fc/v2/pkg/protocols/headless/engine/engine.go#L57-L59

And I can see that the option is passed to chromium as:
```
--proxy-server=http://127.0.0.1:8080
```

Furthermore if I run the same browser with almost all the same options (only `--no-startup-window` removed) then the browser uses the proxy settings without issue.

I'm at a loss as to why the creation of a browser window programatically would disable the proxy settings, but this is the only difference I can think of at the moment.

## Other info

Full command which starts the headless chromium instance
```
└─$ ps -h | rg 13355

  13355 pts/1    Sl     0:00 /home/kali/.cache/rod/browser/chromium-901912/chrome-linux/chrome --disable-sync --enable-automation --disable-ipc-flooding-protection --remote-debugging-port=0 --window-size=1080,1920 --disable-background-timer-throttling --disable-hang-monitor --metrics-recording-only --disable-blink-features=AutomationControlled --disable-notifications=true --no-sandbox=true --user-data-dir=/tmp/nuclei-374078385 --disable-features=site-per-process,TranslateUI --proxy-server=http://127.0.0.1:8080 --disable-breakpad --disable-prompt-on-repost --no-first-run --disable-backgrounding-occluded-windows --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-renderer-backgrounding --enable-features=NetworkService,NetworkServiceInProcess --ignore-certificate-errors=1 --incognito=true --disable-background-networking --disable-default-apps --disable-dev-shm-usage --disable-popup-blocking --disable-gpu=true --disable-crash-reporter=true --mute-audio=true --no-startup-window --force-color-profile=srgb --hide-scrollbars=true --enable-crashpad
```

Chromium version
```
$ /home/kali/.cache/rod/browser/chromium-901912/chrome-linux/chrome --version
Chromium 93.0.4577.0
```

Nuclei version
```
$ nuclei --version

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.5.2

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Current Version: 2.5.2
```

**Comment by ehsandeep** [comment, marked as answer]
@tomdottom thank you for pointing this out, this is now fixed and pushed in the dev version - https://github.com/projectdiscovery/nuclei/issues/1129

**Reply by tomdottom** [reply]
@ehsandeep  this works great with http proxies :+1: 

However, it does not currently work with a socks proxy.

**Reply by ehsandeep** [reply]
@tomdottom can you checkout again with fixed PR and confirm if it's working? https://github.com/projectdiscovery/nuclei/pull/1155

**Reply by ehsandeep** [reply]
should be fixed and merged into the dev branch now.
